src/app/home-page/components/slideshow/slideshow.component.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  Component,
+  ViewChild,
+  ElementRef,
+  OnInit,
+  AfterViewInit,
+  OnDestroy
+} from '@angular/core';
+import { fromEvent, Subscription } from 'rxjs';
+import { throttleTime } from 'rxjs/operators';
+
+enum SLIDE_DIRECTION {
+  RIGHT = 1,
+  LEFT = 0
+}
+
+@Component({
+  selector: 'app-slideshow',
+  templateUrl: './slideshow.component.html',
+  styleUrls: ['./slideshow.component.scss']
+})
+
+export class SlideshowComponent implements OnInit, AfterViewInit, OnDestroy {
+  @ViewChild('allSlides') allSlides: ElementRef;
```

Comment:
rename allSlides to slidesHolder as className . It's more appropriate name

Reply:
done